bindings/pyroot_experimental/PyROOT/python/ROOT/pythonization/_tchain.py

```diff
@@ -0,0 +1,20 @@
+
+from ROOT import pythonization
+
+from ._ttree import _SetBranchAddress as TTreeSetBranchAddress
+
+# Pythonizor function
+@pythonization
+def pythonize_tchain(klass, name):
+    # Parameters:
+    # klass: class to be pythonized
+    # name: string containing the name of the class
+
+    if name == 'TChain':
```

Comment:
Does this check mean that this function may be called with a different class in `name` than `TChain`?

Reply:
I don't understand. If there are 100 classes with pythonizations (i.e. each with a function like the above), when a class is used all 100 functions will be called?

Reply:
That is correct. [This loop](https://github.com/root-project/root/blob/master/bindings/pyroot_experimental/cppyy/CPyCppyy/src/Pythonize.cxx#L825) on the pythonizors gets executed every time a new class is referenced from the application **for the first time**.

Reply:
For every new class, Pythonisations are brushed. This is a good point: maybe we can decide to "discard" a pythonisation once it applied?

Reply:
There is actually a method in cppyy API called [remove_pythonization](https://github.com/root-project/root/blob/master/bindings/pyroot_experimental/cppyy/cppyy/python/cppyy/_pythonization.py#L22), so I guess the pythonizor method can unsubscribe itself if all the conditions it checks (e.g. class name) have applied once.

Reply:
That seems wasteful. Why not have a map so that only one function needs to be run instead of looping over all pythonizor functions each time?

Reply:
I believe the rational followed by Wim to implement it that way is that the condition to check when deciding whether to apply a pythonization can be more complex than just looking at the name of the class. But I agree with your comment and we could have a hybrid solution with per-class pythonizations addressed by a map plus a list of some pythonizations with more complex conditions.